thunder/tests/opinfos.py

```diff
@@ -2607,6 +2605,54 @@ def nan_to_num_error_generator(op, device, dtype=torch.float32, **kwargs):
 conditional_and_mask_ops.append(nan_to_num_opinfo)
 
 
+def all_tensor_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    # when dim is None
+    cases = (
+        (4, 4),
+        (2, 3),
+        (0, 2),
+        (2, 3, 4),
+    )
+
+    for input_shape in cases:
+        yield SampleInput(make(input_shape))
+
+    # input shape, dim, keepdim
+    # dim_cases = (
+    #     ((4, 4), None, False),
+    #     ((4, 4), None, True),
+    #     ((2, 3), 0, True),
+    #     ((2, 3, 4), (1, 2), False),
+    #     ((2, 3, 4), (1, 2), True),
+    #     ((2, 3, 4), (-1, 1), False),
+    #     ((2, 3, 4), (-1, 1), True),
+    # )
+
+    # for input_shape, dim, keepdim in dim_cases:
+    #     yield SampleInput(make(input_shape), dim, keepdim)
+
+
+def all_tensor_error_generator(op, device, dtype=torch.float32, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype)
+    err_msg = r"Dimension out of range \(expected to be in range of \[.*?\], but got .*\)"
+    yield (
+        SampleInput(make(5, 1, 2, 3), 4),
+        IndexError,
+        err_msg,
+    )
+
+
+all_tensor_opinfo = OpInfo(
+    ltorch.all_tensor,
+    sample_input_generator=all_tensor_sample_generator,
+    error_input_generator=all_tensor_error_generator,
+    torch_reference=torch.all,
+)
+conditional_and_mask_ops.append(all_tensor_opinfo)
```

Comment:
Let's add this OpInfo to the reduction opinfos:

https://github.com/k223kim/lightning-thunder/blob/2ee356f56aaadb2d38c2baa3d4b7bbddf9bd674f/thunder/tests/opinfos.py#L4605

`all` isn't like most other reductions, but it is a reduction

Reply:
I sometimes get confused with where to put the test cases. Thank you for advising! Will do that :)